src/server/binary/binaryServerStarter.ts

```diff
@@ -217,6 +217,46 @@ function httpHttpsGet(url: string, callback: (response: http.IncomingMessage) =>
   return http.get(url, callback);
 }
 
+/**
+ * Show a progress notification for the given get response
+ *
+ * Inspired by
+ * https://stackoverflow.com/questions/18323152/get-download-progress-in-node-js-with-request
+ *
+ * @param response the http GET response
+ */
+function showProgressForDownload(response: http.IncomingMessage) {
+  const progressOptions: ProgressOptions = {
+    location: ProgressLocation.Notification,
+    cancellable: true,
+    title: "Downloading the XML Language Server..."
+  };
+  window.withProgress(
+    progressOptions,
+    (progress, cancelToken) => {
+      const size: number = parseInt(response.headers['content-length']);
+      let downloaded: number = 0;
+      let previousReportedDownloadedPercent: number = 0;
+      response.on("data", (chunk) => {
+        downloaded += chunk.length;
+        const incrementAmt = ((100.0 * downloaded / size) - previousReportedDownloadedPercent);
+        previousReportedDownloadedPercent += incrementAmt;
+        progress.report({
+          increment: incrementAmt
+        });
+      });
+      const downloadFinish: Promise<void> = new Promise((resolve, _reject) => {
+        response.on("close", () => {
+          resolve();
+        })
+      });
+      cancelToken.onCancellationRequested((_e: any) => {
+        response.destroy(new Error("Cancelled by user"));
```

Comment:
<img width="558" alt="Screenshot 2021-03-09 at 18 23 05" src="https://user-images.githubusercontent.com/148698/110511458-89ff6300-8104-11eb-8053-f92bfb5d9169.png">

shouldn't it be more appropriate (i.e english-ier) to say: falling back to using the Java Language Server? If current message is ok, then so be it.

Reply:
looks like it's downloaded even when cancelled

Reply:
I think it might partially download the file. I'll test to see. I'll fix the English and add some code to clean up if a cancel is issued.

Reply:
It should properly clean up if the download is cancelled.